pkg/orm/dojo_worker.go

```diff
@@ -33,6 +77,18 @@ func (s *DojoWorkerORM) CreateDojoWorker(walletAddress string, chainId string) (
 }
 
 func (s *DojoWorkerORM) GetDojoWorkerByWalletAddress(walletAddress string) (*db.DojoWorkerModel, error) {
+	workerCache := NewDojoWorkerCache(WorkerByWalletCacheKey)
+	workerCache.walletAddress = walletAddress
```

Comment:
as you are trying to build the cache key, you could actually make NewDojoWorkerCache to accept variadic arguments e.g. https://gobyexample.com/variadic-functions, so that it is also easier later on.
if you split by colons it's also easier to view in medis

```
func NewDojoWorkerCache(keyParts ...string) *DojoWorkerCache {
	key := strings.Join(keyParts, ":")
	return &DojoWorkerCache{
		key: key,
	}
}
```